### PR TITLE
Grid - Columns and Rows

### PR DESCRIPTION
**Pull request by JohnCoene**
Hi,

I added arguments to specify width of columns as well as create resizable columns and rows.

Example:

```r
library(shiny)

shiny::shinyApp(
  ui = f7Page(
    title = "Grid",
    f7SingleLayout(
     navbar = f7Navbar(title = "f7Row, f7Col"),
     f7Row(
     f7Col(
      resizable = TRUE,
      f7Card(
       "This is a simple card with plain text,
       but cards can also contain their own header,
       footer, list view, image, or any other element."
      )
     ),
     f7Col(
      resizable = TRUE,
      f7Card(
       title = "Card header",
       "This is a simple card with plain text,
        but cards can also contain their own header,
        footer, list view, image, or any other element.",
       footer = tagList(
        f7Button(color = "blue", label = "My button", src = "https://www.google.com"),
        f7Badge("Badge", color = "green")
       )
      )
     )
    )
    )
  ),
  server = function(input, output) {}
 )
}
```

Note that it's one or the other; specify size or resizable.

### CODE REVIEW

File: R/f7Grid.R [hunk 1]
```diff
@@ -82,7 +82,7 @@ f7Col <- function(..., width = NULL, resizable = FALSE){
 
   # width
   if(!is.null(width))
-    cl <- paste0(col, width)
+    cl <- paste0(cl, "-", width)
 
   # sizable
   if(resizable) cl <- paste(cl, "resizable")
```
Comment (DivadNojnarg) [inline R/f7Grid.R]: Below I would also do:
```r
if (resizable) col <- shiny::tagAppendAttributes(col, style = sprintf("min-width: %s", param))
```
to consider the minimum allowed width.

**Comment by DivadNojnarg** [comment]
There is also an issue when we resize rows -> the lower row covers the upper one. Seems that something is missing from [here](https://framework7.io/docs/grid.html):

```r
library(shiny)
library(shinyMobile)

mycard <- f7Card(
  "This is a simple card with plain text,
  but cards can also contain their own header,
  footer, list view, image, or any other element."
)

myrow1 <- f7Row(
  f7Col(resizable = TRUE, mycard),
  f7Col(resizable = TRUE, mycard)
)

myrow2 <- f7Row(
  resizable = TRUE,
  f7Col(mycard),
  f7Col(mycard)
)


shiny::shinyApp(
  ui = f7Page(
    title = "Grid",
    f7SingleLayout(
      navbar = f7Navbar(title = "f7Row, f7Col"),
      tagList(myrow1, myrow2, myrow2)
    )
  ),
  server = function(input, output) {}
)
}
```

Screenshot 2020-02-05 at 22 14 17

**Comment by DivadNojnarg** [comment]
Finally we also need to resize plots on `grid:resize` with js:

```javascript
$(function() {
    $('.col, .row').on('grid:resize', function() {
        $(this).trigger('shown');
    });
});
```

also with a debounce policy so that we only trigger the change when we release the handler

**Comment by DivadNojnarg** [comment]
Well, in the framework7 demo, they use custom css to make row resize working. But it would break all other components from shinyMobile. I think it's ok to assume we can only resize columns for now...

File: R/f7Grid.R (hunk 1 above)
Comment (DivadNojnarg) [reply R/f7Grid.R]: Can you add an extra parameter to account for fixed-width columns? We need to add the class "resizable-fixed" after "resizable" if fixed is TRUE

**Comment by JohnCoene** [comment]
I really got it to work. I’ll look again.

**Comment by JohnCoene** [comment]
I have to rework this entirely as it's not going to work as-is. I don't have the time now so will close.

**Pull request closed without merging**